### PR TITLE
Update outdated documentation and add examples

### PR DESCRIPTION
**Pull request by laevandus**
### 🔗 Issue Links

Resolves: [PBE-5540](https://stream-io.atlassian.net/browse/PBE-5540)

### 🎯 Goal

Update outdated content and add more examples

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS)

[PBE-5540]: https://stream-io.atlassian.net/browse/PBE-5540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docusaurus/docs/iOS/client/state-layer/state-layer-overview.md [hunk 1]
```diff
@@ -104,40 +138,82 @@ try await chat.loadOlderMessages()
...
+    limit: 10
+)
+// access all the currently loaded reactions
+let allLoadedReactions35 = messageState.reactions
+```
+
+:::note
+Use `Chat` for loading all the reactions per message and `ReactionList` if you need to load reactions using a filter.
+:::
+
```
Comment (laevandus) [inline docusaurus/docs/iOS/client/state-layer/state-layer-overview.md:216]: WIP, will add more examples

File: docusaurus/docs/iOS/client/state-layer/state-layer-overview.md [hunk 2]
```diff
@@ -230,4 +306,35 @@ var users = userSearch.state.users
 try await userSearch.search(term: "thie")
 // all the fetched results for "thie"
 users = userSearch.state.users
-```
\ No newline at end of file
+```
+
+## Listening to Web-Socket Events
+
+Web-socket events are delivered when-ever server side state changes. Stream chat's low-level client listen to these events and updates the local state accordingly. If there is a need, we can subscribe to these events using `ChatClient` for any events including channel events, and `Chat` for channel specific events. The latter offers convenience over the former.
```
Comment (martinmitrevski) [inline docusaurus/docs/iOS/client/state-layer/state-layer-overview.md:313]: - whenever*
- Stream chat's low-level client listens*

File: docusaurus/docs/iOS/client/state-layer/state-layer-overview.md [hunk 3]
```diff
@@ -104,40 +138,82 @@ try await chat.loadOlderMessages()
 messages = chat.state.messages
 ```
 
-If we would like to observe a single message's state we can use the `MessageState` observable object.
+:::tip Call get with watch true once per app lifetime
```
Comment (martinmitrevski) [inline docusaurus/docs/iOS/client/state-layer/state-layer-overview.md:141]: maybe use `get` and `watch`, reading this without them is a bit confusing.

File: Sources/StreamChat/Controllers/ChannelController/ChannelController.swift [hunk 4]
```diff
@@ -10,6 +10,7 @@ import Foundation
 /// `ChatChannelController` objects are lightweight, and they can be used for both, continuous data change observations (like
 /// getting new messages in the channel), and for quick channel mutations (like adding a member to a channel).
 ///
+/// - Note: Please refer to ``Chat`` in the async-await supported [state layer](https://getstream.io/chat/docs/sdk/ios/client/state-layer/state-layer-overview/).
```
Comment (martinmitrevski) [inline Sources/StreamChat/Controllers/ChannelController/ChannelController.swift:13]: Nice way to promote the state layer. I would rephrase these a bit though - something like "For an async-await alternative of the $ClassName, please check ..."

File: docusaurus/docs/iOS/client/state-layer/state-layer-overview.md [hunk 5]
```diff
@@ -104,40 +138,82 @@ try await chat.loadOlderMessages()
...
+    extraData: [
+      "my-custom-key": .string("and string value")
+    ]
+)
 ```
 
-:::tip Call get with watch true once per app lifetime
-The get method fetches the latest state from the Stream API and if we set watch to true, server-side events will keep your local state up to state.
-:::
+If we would like to create a message thread within a channel, the `Chat` type provides a reply method with all the before-mentioned arguments.
```
Comment (martinmitrevski) [inline docusaurus/docs/iOS/client/state-layer/state-layer-overview.md:164]: same here `reply`

**Review by martinmitrevski — COMMENTED** [review]
Looks great 👍 Just few small things before merging

**Review by martinmitrevski — APPROVED** [review]
